rust/rope/src/lib.rs

```diff
@@ -730,8 +779,9 @@ impl Node {
         (node.get_leaf(), offset)
     }
 
-    // precondition: offset > 0 and in range
     fn prev_codepoint_offset(&self, offset: usize) -> usize {
+        debug_assert!(offset > 0 && offset <= self.len());
```

Comment:
Should this be `offset < self.len()`? (It's entirely possible that it's correct as is)